### DISCUSSION TITLE
Full enum support in mount method

### DISCUSSION THREAD

**Discussion by joshua-frese** [livewire/livewire#4929] [1. Help]
I tried to use Enums. My call was:
`@livewire('test', ['showActions' => [App\Enums\Action::SHOW])`

and the mount method looked like:
```
    /**
     * @param array<App\Enums\Example> $showActions
     * @return void
     */
    public function mount(array $showActions = [App\Enums\Action::SHOW]) {
            $this->showActions = $showActions;
    }
```

Blade:
```
   @if (in_array(\App\Enums\Action::SHOW, $showActions))
            <x-jet-button>Show</x-jet-button>
    @endif
```

Enum:
```
enum Action
{
    case SHOW;
}
```
The initial call worked (F5), but when I go on the second page with the "pagination" logic, the livewire component lost the enum information.

My workaround, with enum, is the following example:
`@livewire('test')`

```
    /**
     * @param array<App\Enums\Example::{}->value> $showActions
     * @return void
     */
    public function mount(?array $showActions = null) {
        if ($showActions === null) {
            $this->showActions = [\App\Enums\Action::SHOW->value];
        } else {
            $this->showActions = $showActions;
        }
    }
```

Blade:
```
   @if (in_array(\App\Enums\Action::SHOW->value, $showActions))
            <x-jet-button>Show</x-jet-button>
    @endif
```

Enum:
```
enum Action: string
{
    case SHOW = 'show';
}
```

Or it is a bug? No Idea, I have a good workaround but my initial idea was a little bit better

**Comment by joshhanley** [comment]
@etjavis one of the last couple of versions of Livewire added support for Enums as component properties, so make sure you upgrade to the latest version. See this PR https://github.com/livewire/livewire/pull/4506

But don't make it an array, make sure you store the enum directly in a property and it should work.

Hope this helps!
